asf_heat_pump_suitability/pipeline/transform/non_residential_entities.py

```
@@ -27,29 +29,42 @@
     important_building_gdf: gpd.GeoDataFrame,
     railway_station_gdf: gpd.GeoDataFrame,
+    poi_gdf: gpd.GeoDataFrame,
     building_gdf: gpd.GeoDataFrame,
+    uprns_gdf: gpd.GeoDataFrame,
 ) -> gpd.GeoDataFrame:
     """
-    Transform important buildings and railway data to create a dataframe of polygons representing buildings
-    which are unlikely to contain residential properties, e.g. hospitals, train stations, museums etc.
+    Use important buildings, railway station, and points of interest data to create a dataframe of polygons representing buildings
+    which are unlikely to contain residential properties, e.g. hospitals, train stations, museums etc. Important buildings
+    and POI geopoints should include only types which are unlikely to be in mixed-use (residential and commercial) buildings.
 
     Args:
         important_building_gdf (gpd.GeoDataFrame): OS OpenMap Local important building footprints in area of interest
         railway_station_gdf (gpd.GeoDataFrame): OS OpenMap Local railway station point geometries in area of interest
+        poi_gdf (gpd.GeoDataFrame): non-domestic Points of Interest geopoints in area of interest
         building_gdf (gpd.GeoDataFrame): all building footprints in area of interest
+        uprns_gdf (gpd.GeoDataFrame): UPRNs with point geometries in area of interest
 
     Returns:
         gpd.GeoDataFrame: geometries of buildings which are unlikely to contain residential properties
     """
     print("Creating non-residential buildings dataset...")
     # Assert all gdfs have the same CRS
     assert (
-        len({important_building_gdf.crs, railway_station_gdf.crs, building_gdf.crs})
+        len(
+            {
+                important_building_gdf.crs,
+                railway_station_gdf.crs,
+                poi_gdf.crs,
+                building_gdf.crs,
+                uprns_gdf.crs,
+            }
+        )
         == 1
-    )
+    ), "All GeoDataFrame inputs must have the same CRS"
```

as per the previous PR, we might want to raise an exception?

---

I think this is enough. It will error out and then a user would see this error message.